src/source/parser.scala

```diff
@@ -29,31 +29,50 @@ import scala.collection.mutable
 import scala.util.parsing.combinator.RegexParsers
 import scala.util.parsing.input.{Position, Positional}
 
-case class Parser() {
+case class Parser(includePaths: List[String]) {
 
 val visitedFiles = mutable.Set[File]()
 val fileStack = mutable.Stack[File]()
 
 private object IdlParser extends RegexParsers {
   override protected val whiteSpace = """[ \t\n\r]+""".r
 
-  def idlFile(origin: String): Parser[IdlFile] = rep(importFile) ~ rep(typeDecl(origin)) ^^ { case imp~types => IdlFile(imp, types) }
-
-  def importFile: Parser[FileRef] = {
-    
-	def fileParent:String = if (fileStack.top.getParent() != null) return fileStack.top.getParent() + "/" else return ""
+  def idlFile(origin: String): Parser[IdlFile] = rep(importFileRef) ~ rep(typeDecl(origin)) ^^ { case imp~types => IdlFile(imp, types) }
 
+  def importFileRef(): Parser[FileRef] = {
     ("@" ~> directive) ~ ("\"" ~> filePath <~ "\"") ^^ {
-      case "import" ~ x =>
-        val newPath = fileParent + x
-        new IdlFileRef(new File(newPath))
-      case "extern" ~ x =>
-        val newPath = fileParent + x
-        new ExternFileRef(new File(newPath))
+      case "import" ~ x => {
+        new IdlFileRef(importFile(x))
+      }
+      case "extern" ~ x => {
+        new ExternFileRef(importFile(x))
+      }
     }
   }
+
+  def importFile(fileName: String): File = {
+    var file: Option[File] = None
+
+    includePaths.foreach(path => {
+      if (file.isEmpty) {
+        var relPath = if (path == ".") fileParent else path + "/"
```

Comment:
See my note elsewhere about this default.  It's the right default behavior, but I think this needs to be a special case, rather than a behavior of ".", because if a user passes "." from the command line it needs to have its traditional meaning of the current working directory, not the behavior here.